### PR TITLE
More API review changes

### PR DESCRIPTION
**Pull request by JamesNK**
* Change EndpointOptions visibility to internal
* Change IRouteNameMetadata to IRouteValuesAddress
* Change metadata namespace
* Remove RequiredValues from MatcherEndpoint
* Implementations of metadata interfaces

https://github.com/aspnet/Routing/issues/676

### CODE REVIEW

File: src/Microsoft.AspNetCore.Routing/CompositeEndpointDataSource.cs [hunk 1]
```diff
@@ -127,9 +126,6 @@ private string DebuggerDisplayString
                         var template = matcherEndpoint.RoutePattern.RawText;
                         template = string.IsNullOrEmpty(template) ? "\"\"" : template;
                         sb.Append(template);
-                        sb.Append(", Required Values: new { ");
-                        sb.Append(string.Join(", ", FormatValues(matcherEndpoint.RequiredValues)));
-                        sb.Append(" }");
                         sb.Append(", Defaults: new { ");
```
Comment (kichalla) [inline src/Microsoft.AspNetCore.Routing/CompositeEndpointDataSource.cs]: Ah, this is needed for display. Could you follow the pattern from lines 136-138

**Comment by JamesNK** [comment]
@rynowak Is IRouteValuesAddressMetadata how you envisaged it? One thing that is a small flag to me is `IRouteValuesAddressMetadata` has Name and RequiredValues while `RouteValueAddress` has Name, ExplicitValues and AmbientValues.

File: src/Microsoft.AspNetCore.Routing/IRouteValuesAddressMetadata.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public interface IRouteValuesAddressMetadata
+    {
+        string Name { get; }
+        IReadOnlyDictionary<string, object> RequiredValues { get; }
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Routing/IRouteValuesAddressMetadata.cs]: Yeah, looks right 👍

**Comment by JamesNK** [comment]
@kichalla Do you still want to make these changes part of a feature branch you're making? This PR is good to go.

**Comment by kichalla** [comment]
Just a reminder...As discussed earlier, I will take care of creating a feature branch for this and finally merging it.

**Comment by kichalla** [comment]
091cb94094a776b53ca6323c4251a38afc5ceef3

**Pull request closed without merging**